fs/sysfs/dir.c

```diff
@@ -56,8 +59,13 @@ int sysfs_create_dir_ns(struct kobject *kobj, const void *ns)
 
 	kobject_get_ownership(kobj, &uid, &gid);
 
+	if (sysfs_restrict)
+		mode = S_IRWXU;
+	else
+		mode = S_IRWXU | S_IRUGO | S_IXUGO;
+
 	kn = kernfs_create_dir_ns(parent, kobject_name(kobj),
-				  S_IRWXU | S_IRUGO | S_IXUGO, uid, gid,
+				  mode, uid, gid,
```

Comment:
What about the call to `kernfs_create_dir_ns()` from `internal_create_group()`?

Reply:
Isn't that to create a subdirectory? The initial restrictions on sysfs would cover those.